requirements/system/user-client/functional.tex

```diff
@@ -0,0 +1,50 @@
+\section{Functional Requirements}
+\begin{itemize}
+  \item[UCF10] The user client shall implement all relevant command line options for the specified in Command Line Interface.
+  \item[UCF20] In order to carry out a request, the user client shall authenticate the user with the central server as follows:
+    \begin{enumerate}
+      \item[UCF21] Establish a \gls{SSH} connection to the server.
+      \item[UCF22] Log in the user to the server, using their credentials.
+        Credentials are either stored in a local file, environment variable or are entered manually by the user.
+      \item[UCF23] Once the user is authenticated, the actual request is sent to the server using the utility program described in CSF40.
+    \end{enumerate}
+
+  \item[UCF01] User client shall enable the user to submit new jobs to the central server using the `add` command.
+    In case the authentification fails, the job shall not be queued and an error message shall be displayed.
+    Otherwise, the new job shall be queued in the central server with all options set to the values specified from the user.
+    After that, the user client shall exit, except if blocking mode is requested.
```

Comment:
Or if the user attaches his command line to the job.

Reply:
What is the difference between `request blocking mode` and `attach cmdline to the job`? I mean, you could later attach your output, but then we're speaking about different instance of the user client, not the one used to add your job.

Reply:
Setting the blocking option to true is functionally the same as setting the attach option to true and setting the verbose option to 0.
The blocking option is essentially just for convenience.